File: cia-core/src/main/java/top/ourisland/creepersiarena/core/command/message/CommandUsage.java
package top.ourisland.creepersiarena.core.command.message;

/**
 * One clickable command usage row in a help panel.
 */
public record CommandUsage(
        String command,
        String description
) {

    public String toMiniRow() {
        var suggest = suggestCommand(command);
        return "<click:suggest_command:'%s'><green>%s</green></click> <dark_gray>-</dark_gray> <gray>%s</gray>".formatted(
                CommandMessenger.escapeForAttribute(suggest),
                CommandMessenger.escape(command),
                CommandMessenger.escape(description)
        );
    }

    private String suggestCommand(String command) {
        if (command == null || command.isBlank()) return "/cia";
        var idx = command.indexOf('<');
        if (idx < 0) return command;
        var prefix = command.substring(0, idx).stripTrailing();
        return prefix.endsWith(" ") ? prefix : prefix + " ";
    }

}

File: cia-core/src/main/java/top/ourisland/creepersiarena/core/command/message/CommandUsage.kt
package top.ourisland.creepersiarena.core.command.message

/**
 * One clickable command usage row in a help panel.
 */
class CommandUsage(
    private val command: String?,
    private val description: String?
) {

    fun command(): String? = command

    fun description(): String? = description

    fun toMiniRow(): String {
        val suggest = suggestCommand(command)
        return "<click:suggest_command:'${CommandMessenger.escapeForAttribute(suggest)}'><green>${
            CommandMessenger.escape(
                command
            )
        }</green></click> <dark_gray>-</dark_gray> <gray>${CommandMessenger.escape(description)}</gray>"
    }

    private fun suggestCommand(command: String?): String {
        if (command.isNullOrBlank()) return "/cia"

        val placeholderIndex = command.indexOf('<')
        if (placeholderIndex < 0) return command

        val prefix = command.substring(0, placeholderIndex).trimEnd()
        return if (prefix.endsWith(" ")) prefix else "$prefix "
    }

}
